Drop glyphicons and replace it by already existing icons in icons.css

hexaltation opened the pull request:
## Context

[Bump bootstrap from 3.4.1 to 5.3.3](https://github.com/gristlabs/grist-core/commit/d036d288dce12a392e2f6ab1fb710645908966f7) lead to a bug.
In v 4.0.0 bootstrap dropped glyphicon usage.
Grist-core code have some occurences of glyphicon.

```bash
$ grep -rin glyphicon
client/lib/koForm.js:337:    this.optionButton("left", dom('span.glyphicon.glyphicon-align-left'),
client/lib/koForm.js:339:    this.optionButton("center", dom('span.glyphicon.glyphicon-align-center'),
client/lib/koForm.js:341:    this.optionButton("right", dom('span.glyphicon.glyphicon-align-right'),
client/lib/koForm.js:482:            dom('span.kf_drag_indicator.glyphicon.glyphicon-option-vertical') :
client/lib/koForm.js:487:            return dom('span.drag_delete.glyphicon.glyphicon-remove',
client/components/DocConfigTab.js:19:      dom('span.glyphicon.glyphicon-check'),
client/components/ValidationPanel.js:66:            dom('div.validation_trash.glyphicon.glyphicon-remove',
client/components/ValidationPanel.js:71:            1, dom('div.glyphicon.glyphicon-tag.config_icon'),
client/components/ColumnFilters.css:69:.g-glyphicon-tristate {
client/components/RecordLayoutEditor.js:128:    dom('div.g_record_delete_field.glyphicon.glyphicon-eye-close',
client/widgets/AttachmentsWidget.ts:193:const cssAttachmentIcon = styled('div.glyphicon.glyphicon-paperclip', `
```

## Proposed solution

Remove Glyphicon calls and replace it by corresponding icons in `static/icon/icons.css`

## Related issues

Fixes #1412

## Has this been tested?

- [ ] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this is not relevant here
- [ ] 🙋 no, because I need help <!-- Detail how we can help you -->

hexaltation commented on app/client/components/ColumnFilters.css:
```diff
@@ -66,15 +66,6 @@
   margin-right: 4px;
 }
 
-.g-glyphicon-tristate {
```
Removed because not used elsewhere

hexaltation commented on app/client/components/DocConfigTab.js:
This file is removed cause not used elsewhere

hexaltation commented on app/client/components/GristDoc.ts:
Remove not really used DocConfigTab

hexaltation commented on app/client/components/ValidationPanel.css:
This file is removed cause not used elsewhere

hexaltation commented on app/client/components/ValidationPanel.js:
This file is removed cause not used elsewhere

hexaltation commented on app/client/lib/multiselect.css:
This file is removed cause not used elsewhere

hexaltation commented on app/client/lib/multiselect.js:
This file is removed cause not used elsewhere

hexaltation commented on app/client/lib/koForm.js:
```diff
@@ -388,237 +371,24 @@ exports.collapsible = function(contentFunc, isMountedCollapsed) {
 };
 
 
-/**
```
This section is removed cause not used elsewhere

georgegevoian replied:
This is still used by `app/client/widgets/ConditionalStyle.ts`.

hexaltation commented on app/client/lib/koForm.js:
```diff
@@ -329,20 +326,6 @@ exports.selectSpinner = function(valueObservable, optionObservable) {
   return genSpinner(valueObservable, getNewValue, shouldDisable);
 };
 
-/**
```
This section is removed cause not used anywhere

georgegevoian commented on app/client/components/DocConfigTab.js:
Could you also remove `declare module "app/client/components/DocConfigTab";` from `app/client/declarations.d.ts`? Thanks.

georgegevoian commented on app/client/components/RecordLayoutEditor.js:
```diff
@@ -151,4 +151,18 @@ const cssCollapseIcon = styled(icon, `
   margin: -3px -2px -2px 2px;
 `);
 
+const cssIconEyeClose = styled('div.g_record_delete_field', `
```
Is there some way to reuse `icon` from `app/client/ui2018/icons.ts` here and in `AttachmentsWidget.ts`?

hexaltation replied:
It's the end of the day for me and i'm off tomorrow.
If you want a quick fix for self hosters it's more than ok that you add the last fixes.
Else i'll work on it saturday or monday.

georgegevoian replied:
The remaining stuff is minor and getting a fix out ASAP is more important, so we'll merge now as is. Thanks @hexaltation.

paulfitz commented:
Related report: https://community.getgrist.com/t/bug-in-my-self-hosted-instance-no-paperclip-icon-on-attachment-field/8222/2

hexaltation commented on app/client/lib/koForm.css:
It renders like this
![grist_draggable list](https://github.com/user-attachments/assets/2af0f413-21c6-4e4b-9a81-e106a8b9eff7)

hexaltation commented on app/client/lib/koForm.js:
```diff
@@ -479,12 +465,12 @@ exports.draggableList = function(contentArray, itemCreateFunc, options) {
           (options.drag_indicator ?
            (typeof options.drag_indicator === 'boolean' ?
-            dom('span.kf_drag_indicator.glyphicon.glyphicon-option-vertical') :
+            dom('span.kf_drag_indicator.icon-dragdrop') :
```
`drag_indicator` is set to cssDragger everywhere in the app, so it's not really used.
But the fallback still exists. See koForm.css for a screenshot of the render.

hexaltation commented on app/client/lib/koForm.js:
```diff
@@ -479,12 +465,12 @@ exports.draggableList = function(contentArray, itemCreateFunc, options) {
           kd.domData('model', item),
           kd.maybe(removeFunc !== undefined && options.removeButton, function() {
-            return dom('span.drag_delete.glyphicon.glyphicon-remove',
+            return dom('span.drag_delete.icon-remove',
```
`RemoveFunc` seems to not being used in the app
But the fallback still exists. See koForm.css for a screenshot of the render.

georgegevoian commented:
For the most recent browser test failure, you can just remove the failing test (related to the now-removed validation feature). If you grep the codebase for `validationsTool`, there's a few other small things you can remove as well.

paulfitz commented:
Haven't looked at code, but not seeing paperclip in the preview?

![Screenshot from 2025-02-06 10-35-06](https://github.com/user-attachments/assets/892e0349-e84b-4279-9c7c-793b5828e5ac)

georgegevoian commented on app/client/lib/koForm.css:
```diff
@@ -203,6 +214,17 @@ div:hover > .kf_tooltip {
   color: #777777;
 }
 
+.drag_delete.icon-remove::before {
+  display: block;
+  background-color: var(--grist-theme-control-secondary-fg, var(--grist-color-slate));
+  content: ' ';
+  mask-image: var(--icon-Remove);
+  width: 14px;
+  height: 14px;
+  mask-size: contain;
+  mask-repeat: no-repeat;
+}
+
```
Since this mostly shares the same styles as above, maybe we can break out the common stuff into a new `.kf_draggable__icon` rule?

hexaltation commented on app/client/widgets/AttachmentsWidget.ts:
```diff
@@ -253,4 +253,15 @@ const cssFileType = styled('div', `
+    height: 14px;
+    mask-size: contain;
+    mask-repeat: no-repeat;
+  }
```
Thanks
moved in 5d046283

hexaltation commented:
> For the most recent browser test failure, you can just remove the failing test (related to the now-removed validation feature). If you grep the codebase for `validationsTool`, there's a few other small things you can remove as well.

Thanks 
it's now cleaned

georgegevoian reviewed (approved):
Thanks @hexaltation.
